src/components/popup/details/details.jsx

```diff
@@ -28,24 +28,28 @@ export default class Details extends Component {
 		return Object.values(vendors);
 	};
 
-	filterVendors = ({ isCustom = null, purposeIds = [], featureIds = [] } = {}) => {
-		return this.getVendors().filter(vendor => {
-			if (isCustom !== null && (isCustom && vendor.globalId || !isCustom && !vendor.globalId)) {
-				return false;
+	filterVendors = ({ isCustom = null, purposeIds = [], featureIds = [], specialPurpose=null, specialFeature=null } = {}) => {
+		const { gvl } = this.props.store.tcModel;
+		let vendList;
+
+		if(!!purposeIds.length) {
+			if(specialPurpose) {
+				vendList = Object.values(gvl.getVendorsWithSpecialPurpose(purposeIds))
+			} else {
+				vendList = Object.values(gvl.getVendorsWithConsentPurpose(purposeIds))
 			}
-
-			const vendorPurposeIds = new Set([...(vendor.purposes || []), ...(vendor.legIntPurposes || [])]);
-			if (!purposeIds.every(purposeId => vendorPurposeIds.has(purposeId))) {
-				return false;
-			}
-
-			const vendorFeatureIds = new Set(vendor.features|| []);
-			if (!featureIds.every(featureId => vendorFeatureIds.has(featureId))) {
-				return false;
+		} else if(!!featureIds.length) {
+			if(specialFeature) {
+				vendList = Object.values(gvl.getVendorsWithSpecialFeature(featureIds))
+			} else {
+				vendList = Object.values(gvl.getVendorsWithFeature(featureIds))
 			}
+		}
 
-			return true;
-		});
+		if(isCustom) {
+			vendList = vendList.filter(vendor => !!vendor.globalId)
+		}
+		return vendList;
```

Comment:
From:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
callback - Function is a predicate, to test each element of the array. Return true to keep the element, false otherwise.